applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_small_strain_u_pw_element.cpp

```diff
@@ -0,0 +1,545 @@
+{
+    auto& r_model_part = CreateModelPartWithUPwSolutionStepVariables(rModel);
+
+    PointerVector<Node> nodes;
+    nodes.push_back(r_model_part.CreateNewNode(1, 0.0, 0.0, 0.0));
+    nodes.push_back(r_model_part.CreateNewNode(2, 1.0, 0.0, 0.0));
+    nodes.push_back(r_model_part.CreateNewNode(3, 1.0, 1.0, 0.0));
+    const auto p_geometry = std::make_shared<Triangle2D3<Node>>(nodes);
+    return CreateUPwSmallStrainElementWithUPwDofs<2, 3>(rProperties, p_geometry);
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+using namespace Kratos;
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_CreateInstanceWithGeometryInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_geometry =
+        std::make_shared<Triangle2D3<Node>>(ModelSetupUtilities::Create2D3NTriangleGeometry());
+    const auto                        p_properties = std::make_shared<Properties>();
+    const UPwSmallStrainElement<2, 3> element(0, p_geometry, p_properties,
+                                              std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_DoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    Model model;
+    const auto p_element = CreateUPwSmallStrainElementWithUPwDofs(model, std::make_shared<Properties>());
+
+    // Act
+    Element::DofsVectorType degrees_of_freedom;
+    p_element->GetDofList(degrees_of_freedom, ProcessInfo{});
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 9);
+    std::vector<std::string> variable_names;
+    std::transform(degrees_of_freedom.cbegin(), degrees_of_freedom.cend(), std::back_inserter(variable_names),
+                   [](const auto& rpDof) { return rpDof->GetVariable().Name(); });
+    const std::vector<std::string> desired_variable_list{
+        "DISPLACEMENT_X", "DISPLACEMENT_Y", "DISPLACEMENT_X", "DISPLACEMENT_Y", "DISPLACEMENT_X",
+        "DISPLACEMENT_Y", "WATER_PRESSURE", "WATER_PRESSURE", "WATER_PRESSURE"};
+
+    KRATOS_EXPECT_EQ(variable_names, desired_variable_list);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_IntegrationMethod, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_geometry =
+        std::make_shared<Triangle2D3<Node>>(ModelSetupUtilities::Create2D3NTriangleGeometry());
+    const auto                        p_properties = std::make_shared<Properties>();
+    const UPwSmallStrainElement<2, 3> element(0, p_geometry, p_properties,
+                                              std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_integration_method = element.GetIntegrationMethod();
```

Comment:
`GetIntegrationMethod` returns an instance of `IntegrationMethod`, not a pointer to something. So we shouldn't use the `p_` prefix:
```suggestion
    const auto integration_method = element.GetIntegrationMethod();
```

Reply:
removed following the next comment